src/main/java/com/tterrag/k9/commands/CommandCurseGradle.java

```diff
@@ -0,0 +1,309 @@
+package com.tterrag.k9.commands;
+
+import java.awt.Color;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Random;
+
+import com.tterrag.k9.commands.api.Argument;
+import com.tterrag.k9.commands.api.Command;
+import com.tterrag.k9.commands.api.CommandBase;
+import com.tterrag.k9.commands.api.CommandContext;
+import com.tterrag.k9.commands.api.CommandException;
+import com.tterrag.k9.commands.api.Flag;
+
+import lombok.extern.slf4j.Slf4j;
+import sx.blah.discord.handle.obj.IMessage;
+import sx.blah.discord.util.EmbedBuilder;
+import sx.blah.discord.util.RateLimitException;
+import sx.blah.discord.util.RequestBuffer;
+
+/**
+ * Command to figure out curseforges gradle and dependencies.
+ * @author Wyn Price
+ *
+ */
+@Command
+@Slf4j
+public class CommandCurseGradle extends CommandBase {
+	
+	private class CurseResult 
+	{
+		private final String URL;
+		private final String gradle;
+		
+		public CurseResult(String URL, String gradle) 
+		{
+			this.URL = URL;
+			this.gradle = gradle;
+		}
+		
+		public String getGradle() {
+			return gradle;
+		}
+		
+		public String getURL() {
+			return URL;
+		}
+		
+		@Override
+		public String toString() {
+			return "URL:" + URL + "\nGRADLE: " + gradle + "\n";
+		}
+	}
+	
+    private static final Argument<String> ARG_FILEURL = new WordArgument("fileurl", "The files full URL", true);
+	
+	private static final Flag FLAG_USE_OPTIONAL = new SimpleFlag('o', "optional", "Also gets the gradle files for Optional Files", false);
```

Comment:
Suggested description: "Appends optional project dependencies to the output list."